### ISSUE TITLE
[Intermittent] "Firefox Survey" door hanger is wrongly triggered when first visiting www.fox.com website

### ISSUE THREAD

**Issue by SoftVision-CarmenFat** [mozilla/CookieRestrictionsShield#79]
[Affected versions]:
- Beta 63.0b4
- Cookie Restrictions 1.0.3

[Affected Platforms]:
- All Windows

[Prerequisites]:
- The `shieldStudy.logLevel` string pref with All value was added on a new profile.
- The `extensions.cookie-restrictions_shield_mozilla_org.test.variationName` string pref was added on the same profile with CookiesBlocked value.
- Cookie Restrictions .xpi is installed ([link](https://bugzilla.mozilla.org/show_bug.cgi?id=1485518#c28)). 

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites and navigate to www.fox.com website.
2. Wait for the website to finish loading and observe what happens next.

[Expected result]:
- "Firefox Survey" door hanger is not triggered and displayed.

[Actual result]:
- "Firefox Survey" door hanger is triggered and displayed.

[Additional Notes]:
- Here is the Browser Console warning that might be responsible for triggering the Firefox Survey door hanger from the first visit of the website:
`The page was reloaded because the character encoding declaration of the HTML document was not found when prescanning the first 1024 bytes of the file. The encoding declaration needs to be moved to be within the first 1024 bytes of the file.`

![load ping fox](https://user-images.githubusercontent.com/26000308/45366179-eb532380-b5e6-11e8-8468-aaab3f542b0a.png)

- Added a screen recording the issue:

![fox](https://user-images.githubusercontent.com/26000308/45366266-19d0fe80-b5e7-11e8-9b4f-19bb14f8d175.gif)

**Comment by TanviHacks** [comment]
This is an issue with fox.com itself where the page is autorefreshing itself.

**Comment by TanviHacks** [comment]
One way to mitigate this issue is by checking if the user has interacted with a document before showing the doorhanger.  If the user has never interacted with the document and they refresh because it is loading slowly or broken, it likely has nothing to do with Cookie Restrictions.  Cookie Restrictions are more likely to break tasks that a user is working on or authenticated content.  Hence, I think requiring user interaction before refresh shows the doorhanger makes sense.

**Comment by TanviHacks** [comment]
After discussing with team, we decided this is an edge case which we will leave with since none of the potential solutions are great or full proof.  Marking as P3.

**Comment by SoftVision-CarmenFat** [comment]
We verified this issue using the latest CookieRestrictions v.1.0.5 add-on and it's no longer reproducible.

After first navigation to "www.fox.com", the door hanger is no longer displayed and the "page_reloaded_survey" has the following value: "1".  The door hanger is also not triggered on the following reloads of the page. In this case, the value of  "page_reloaded_survey" changes to "6".

Tested on Windows 10 x64 and Windows 7 x32.
